### ISSUE TITLE
Workflow retries are broken for JavaSDK

### ISSUE THREAD

**Issue by Spikhalskiy** [temporalio/temporal#1944]
## Expected Behavior

[`WorkflowRetryTest`](https://github.com/temporalio/sdk-java/blob/5d8734be38648c9c1837832fb4fc424f26136ecb/temporal-sdk/src/test/java/io/temporal/workflow/WorkflowRetryTest.javarl) and other tests that trigger workflow retry in JavaSDK pass on the temporaliotest/auto-setup:latest image.

## Actual Behavior

The tests pass on temporalio/auto-setup:1.12.1 but not on temporaliotest/auto-setup:latest

## Steps to Reproduce the Problem

1. Start Docker Compose

https://github.com/temporalio/docker-compose/blob/main/docker-compose-cas.yml
Replace `temporalio/auto-setup:1.12.1` with `temporaliotest/auto-setup:latest`
```
docker-compose -f docker-compose-cas.yml up
```

2. Run `WorkflowRetryTest` using an external dockerized Temporal server

```
cd java-sdk
export USE_DOCKER_SERVICE=true
./gradlew :temporal-sdk:test --tests=io.temporal.workflow.WorkflowRetryTest
```

## Specifications

  - Version: 8b71d20d94001ab86b368beb0eeefb900e633f5c

**Comment by Spikhalskiy** [comment]
Most likely related to #1866

**Comment by dnr** [comment]
Fixed by #1949
